pkg/proxy/request_test.go

```diff
@@ -429,3 +429,96 @@ func TestGetDuration(t *testing.T) {
 		assert.Equal(t, scenario.expected, actual)
 	}
 }
+
+func TestGetImage(t *testing.T) {
+	for _, ti := range []struct {
+		tag                   string
+		role                  string
+		imageName             string
+		imageTag              string
+		expectedImageResponse string
+		expectedTagResponse   string
+		expectError           bool
+	}{
+		{
+			tag:                   "valid master image",
+			role:                  "masterImage",
+			imageName:             "hellofresh/kangal-jmeter-master",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-master",
+			expectedTagResponse:   "latest",
+			expectError:           false,
+		},
+		{
+			tag:                   "valid worker set",
+			role:                  "workerImage",
+			imageName:             "hellofresh/kangal-jmeter-worker",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           false,
+		},
+		{
+			tag:                   "valid empty master image and tag",
+			role:                  "masterImage",
+			imageName:             "",
+			imageTag:              "",
+			expectedImageResponse: "",
+			expectedTagResponse:   "",
+			expectError:           false,
+		},
+		{
+			tag:                   "invalid master image name",
+			role:                  "masterImage",
+			imageName:             "hellofresh/kangal-jmeter-master",
+			imageTag:              "latest",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           true,
+		},
+		{
+			tag:                   "invalid worker image tag",
+			role:                  "workerImage",
+			imageName:             "hellofresh/kangal-jmeter-worker",
+			imageTag:              "1.0",
+			expectedImageResponse: "hellofresh/kangal-jmeter-worker",
+			expectedTagResponse:   "latest",
+			expectError:           true,
+		},
+	} {
+		t.Run(ti.tag, func(t *testing.T) {
+
+			image := apisLoadTestV1.ImageDetails{
+				Image: "",
+				Tag:   "",
+			}
+
+			err := error(nil)
+
+			sentImage := ti.imageName + ":" + ti.imageTag
+			expectedImage := ti.expectedImageResponse + ":" + ti.expectedTagResponse
+
+			if ti.role == "masterImage" {
+				request := buildMocFormReq(t, map[string]string{}, "1", string(apisLoadTestV1.LoadTestTypeJMeter), "", sentImage, "")
+				image, err = getImage(request, ti.role)
+			}
+			if ti.role == "workerImage" {
+				request := buildMocFormReq(t, map[string]string{}, "1", string(apisLoadTestV1.LoadTestTypeJMeter), "", "", sentImage)
+				image, err = getImage(request, ti.role)
+			}
+
+			actualImage := image.Image + ":" + image.Tag
+
+			assert.Nil(t, err)
+
+			if ti.expectError {
+				// assert.Error(t, err)
+				assert.NotEqual(t, expectedImage, actualImage)
+			} else {
+				// assert.NoError(t, err)
```

Comment:
Is it a leftover?

Reply:
No. There are two checks there:

- Line 512: Assert getImage() returns no error

- From Line 512 - 520: Check that the value returned by getImage() is equal to the expected value or not.

Reply:
I believe it was referring to the commented assertions. But this should not be necessary anymore, as there's no error output from the `getImage` function after the last changes

Reply:
Completely right. In fact it's removed in the last commit

Reply:
Fixed!